### PR TITLE
feat(avalonia): in-app patch2 Initialize/Update via Patch2GitService (#1817)

### LINKED ISSUE #1817: Avalonia: Update Checker is a non-functional stub — no in-app patch2 Initialize/Update (Method A); Options only sets remote URL
## Summary

Verifying the "Method A — in‑app button" workaround from #1808: in the **Avalonia** GUI there is **no in‑app way to fetch/update the patch2 database at all**. Method A does not exist here.

## What the code actually does

- **Update Checker is a stub.** `Tools → Update Checker` opens `ToolUpdateDialogView`, but the view/VM only show an "Update Checker" title and an *Address* label — no update check, no **Git Patch2**, no clone/update:
  - `FEBuilderGBA.Avalonia/Views/ToolUpdateDialogView.axaml` (title + `AddrLabel` only)
  - `FEBuilderGBA.Avalonia/ViewModels/ToolUpdateDialogViewModel.cs` (`LoadList()` returns a single `"Update Checker"` placeholder row)
  - opened from `FEBuilderGBA.Avalonia/Views/MainWindow.axaml.cs:2518` (`wm.Open<ToolUpdateDialogView>()`); listed as a stub in `Services/ListParityHelper.cs:362`.
- **Options can set the remote URL but cannot clone or fetch.** `OptionsView` has *Patch2 / FE‑Repo / FE‑Repo‑Music Remote URL* textboxes (`OptionsView.axaml:38‑43`), and on OK it calls `ApplySubmoduleRemotes` → `ApplyOneRemote` → `GitUtil.SetSubmoduleRemote(...)` (`OptionsViewModel.cs:266‑289`). `SetSubmoduleRemote` only runs `git remote set-url` on an **existing** repo — it never `git clone`s a missing `config/patch2/`, and there is no fetch/reset. On a fresh extract (no `.git` in `config/patch2/`) it just logs a failure.
- `GitUtil.Clone` / `GitUtil.Update` exist in Core (`FEBuilderGBA.Core/GitUtil.cs:168,183`) and are wired in WinForms, but **nothing in Avalonia calls them**.

## Impact

An Avalonia user with an empty `config/patch2/` (the #1808 scenario) cannot initialize or update the patch database from within the app by any means — the Update Checker is inert and Options is URL‑only. They must fall back to the manual Git / download‑ZIP workarounds (Methods B–D in #1808).

## Expected

Avalonia needs real in‑app **Initialize Repository** (clone) and **Update Repository** (fetch + reset) actions wired to `GitUtil.Clone`/`GitUtil.Update`, surfaced next to the Settings URL fields and in the Patch Manager — the Avalonia half of #1812. (Separately, the Update Checker stub should either check for app updates or be removed to avoid implying functionality it lacks.)

## Scope

**Avalonia** (missing implementation / stub). Blocking/necessary infra.

## Related

- Discussion (with user workaround): #1808
- Empty‑patch2 Patch Manager error: #1811
- Umbrella solution (Init/Update buttons in Settings + Patch Manager, both GUIs): #1812
- FE‑Repo / FE‑Repo‑Midi management: #1813
- WinForms counterpart gap: #1816

Copilot CLI: 1.0.69-0
Model: Claude Opus 4.8 (claude-opus-4.8)

### PR DESCRIPTION
**Pull request by laqieer**
## Summary

Adds real **in-app Initialize (clone) / Update (fetch+reset)** of the git-delivered `config/patch2` database to the **Avalonia** GUI — the Avalonia half of #1812. Previously `GitUtil.Clone`/`GitUtil.Update` existed in Core and were wired only in WinForms, so an Avalonia user with an empty `config/patch2` (the #1808/#1811 fresh-download state) had **no in-app way** to fetch or update the patch database; the Update Checker was an inert stub and Options only set the remote URL.

Plan + cross-model plan board: issue #1817 (all three models BLOCKed the first plan; every required fix was adopted — see the `## Cross-Model Review Board` comment there).

## Scope (per the dual-GUI policy)

New feature → **Avalonia only**; WinForms (stable) is untouched. patch2 only — FE-Repo/FE-Repo-Midi = #1813, first-run wizard = #1814, app update-checker = #1804.

## What changed

- **Core `Patch2GitService`** — reusable, unit-tested clone-or-update-with-backup service lifting the proven WinForms `AutoUpdatePatch2Git` decision out of the GUI. Single-flight guard (both entry points), moves **any** existing dir aside before clone (incl. an empty submodule placeholder), unique+guarded backup path, restore-on-failure. Injectable seam for offline deterministic tests.
- **Patch Manager** — always-visible **Initialize / Update Patch Database** button (label derived from the repo-root `config/patch2` git status via `GetPatch2Dir(CoreState.BaseDirectory)`), off-thread with ~150 ms-throttled progress, synchronous disable + `try/finally` re-enable, refreshes the list and shows an accurate restart-recommended message on success.
- **Options** — **Initialize / Update Patch Database Now** button next to the Patch2 URL; passes the textbox URL directly (`urlOverride`) and persists **only** `submodule_patch2_url`.
- **Update Checker stub** — honest text pointing users to the Patch Manager / Options (app updater tracked in #1804), instead of implying update functionality it lacks.

## Screenshots

Avalonia Patch Manager with an empty `config/patch2` — the new **Initialize Patch Database** button in the left panel (rendered with the real desktop Skia rasterizer via `--screenshot-window=PatchManagerView`):

Patch Manager Initialize/Update

## GUI Test Report

| # | GUI check | Result |
|---|---|---|
| 1 | Patch Manager renders the new **Initialize Patch Database** button in the empty-state (real Skia render, screenshot above) | ✅ Pass |
| 2 | `PatchManagerView` exposes `InitUpdatePatch2Button` headlessly with non-empty content (`Patch2InitUpdateViewTests`) | ✅ Pass |
| 3 | `OptionsView` exposes `InitUpdatePatch2Button` headlessly (`Patch2InitUpdateViewTests`) | ✅ Pass |
| 4 | Button label = "Initialize Patch Database" (non-repo `config/patch2`) / "Update Patch Database" (git repo), via repo-root base dir (`Patch2InitUpdateButtonTests`) | ✅ Pass |
| 5 | No regression across PatchManager/Options/ToolUpdateDialog/ControlProperty/AndroidResource views (90 tests) | ✅ Pass |

## Test plan

- [x] `dotnet build FEBuilderGBA.Core` (Release) — 0 errors
- [x] `dotnet build FEBuilderGBA.Avalonia` (Release) — 0 errors
- [x] Core `Patch2GitServiceTests` — **10/10 passed** (GitNotFound; existing-repo→update + update-failure; non-existent→clone; pre-existing **empty** dir moved-aside→clone success **and** clone-fail restore; non-repo-files replace-on-success **and** restore-on-fail; single-flight → AlreadyRunning; `GetPatch2Dir`)
- [x] Avalonia `Patch2InitUpdateButtonTests` + `Patch2InitUpdateViewTests` — **4/4 passed**
- [x] Regression: `PatchManager|Options|ToolUpdateDialog|ControlProperty|AndroidResource` — **90/90 passed**
- [x] Structural: `AvaloniaScrollViewer` source-scan — **3/3 passed**
- [x] Real GUI screenshot rendered via `--screenshot-window=PatchManagerView` (desktop Skia), shown above

Closes #1817

Copilot CLI: 1.0.69-0
Model: Claude Opus 4.8 (claude-opus-4.8)

### CODE REVIEW

**Review by laqieer — COMMENTED** [review]
## Cross-Model Review Board: APPROVED

I've reviewed PR #1835 against the requirements for #1817 and the established Avalonia/Core architecture. The implementation flawlessly executes the plan.

### Verification Checklist:
1. **InitializeOrUpdateCore**: Correctly handles `gitExe` resolution, branches to update for existing repos, cleanly moves ANY existing directory (including empty ones from non-recursive superproject clones) to a unique `_patch2_backup_<ticks>` path. Restores on failure and deletes backup on success. `GitNotFound` gracefully handled.
2. **Single-flight Guard**: `TryEnter`/`Exit` pattern correctly implemented in `InitializeOrUpdate`, returning `AlreadyRunning`. `InitializeOrUpdateCore` remains perfectly pure, enabling parallel test execution.
3. **Patch Manager Base Directory**: `Patch2ButtonText` correctly derives from the repository root `CoreState.BaseDirectory`, avoiding the per-version subfolder pitfall.
4. **UI Threading & Throttling**: Button correctly disabled synchronously, operations dispatched via `Task.Run`, progress updates efficiently throttled to ~150ms using `Environment.TickCount64` + `Interlocked`, and finally block ensures the UI remains responsive.
5. **Options URL Persisting**: `urlOverride` injected safely, and importantly, only `submodule_patch2_url` is committed to `CoreState.Config` directly avoiding dirty saves of the rest of the options menu.
6. **Update Checker**: Inert stub updated to provide honest, helpful redirection.
7. **Tests**: Excellent offline test seam implementation via delegates (`CloneOp`, `UpdateOp`). All tests are well-isolated using GUID temp directories and rigorously exercise edge cases like clone-fail restoration and single-flight blocking.
8. **WinForms Policy**: Strictly observed. WinForms GUI untouched. The PR screenshot rendering hook is appropriately scoped and harmless.
9. **PR Requirements**: Screenshot from `master` branch present, GUI test report complete, and test plan fully executed.

Excellent work extracting this logic cleanly to `FEBuilderGBA.Core` and building a responsive, bulletproof Avalonia UI around it. No threading, logic, or state-management bugs found.

Copilot CLI: 1.0.69-0
Model: claude-opus-4.8

**Review by laqieer — COMMENTED** [review]
Verdict: REQUEST CHANGES

## Issue: Clone exceptions leave config/patch2 moved aside instead of restored
**File:** FEBuilderGBA.Core/Patch2GitService.cs:126
**Severity:** High
**Problem:** The clone path moves any existing `config/patch2` to `_patch2_backup_<ticks>` before invoking `cloneOp`, but the restore path only runs when `cloneOp` returns a non-zero exit code. If `GitUtil.Clone` throws after the move (for example, `Process.Start`/IO failures after `FindGitExecutable` has already succeeded, or any unexpected exception in the injected operation), control skips lines 130-139 entirely and the public UI catch only reports an error. The user's original patch database is left hidden under the backup name, with `config/patch2` missing or partially created.
**Evidence:** In `Patch2GitService.InitializeOrUpdateCore`, the existing directory is moved at line 126, `cloneOp(...)` is called outside any restore-protecting `try` at line 129, and restoration is only inside the `if (cloneCode != 0)` block at lines 130-139. The new tests cover non-zero clone failures, but not a throwing clone operation, so this rollback hole is currently unguarded.
**Suggested fix:** Wrap the clone invocation and failure handling so all clone failures, including exceptions, go through the same restore path before returning/throwing. Add an offline unit test with an injected `cloneOp` that throws after the backup move and assert the original `config/patch2` is restored and no `_patch2_backup_*` remains.

Additional check: the PR screenshot URL is a default-branch raw URL as required, but it still returned 404 at the end of this review.

Copilot CLI: 1.0.69-0
Model: ChatGPT (exact Copilot CLI model id not exposed in this session)

**Review by laqieer — COMMENTED** [review]
## Verdict: REQUEST CHANGES (BLOCK)

*(Posted as a `--comment` review, not `--request-changes`, because GitHub rejects `addPullRequestReview` approve/changes-requested states on one's own PR — this account authored both the PR and this review. The verdict below is nonetheless a hard REQUEST CHANGES per the board's pessimistic-veto rule.)*

Reviewed at head commit `12abf019de0954c731c7c5622972360477626dd3` in a fresh, isolated worktree (`git worktree add --detach` at that exact SHA, not a possibly-dirty checkout). Overall this is a well-scoped, carefully implemented feature with strong test coverage — I found one concrete correctness gap against the plan's own "restore-on-failure" guarantee that should be closed before merge. Everything else checks out.

### 1. Correctness

Verified correct:
- Backs up **any** existing `config/patch2` dir, including an empty submodule placeholder, before cloning — not just non-empty dirs (`FEBuilderGBA.Core/Patch2GitService.cs:114-116`).
- Stale-backup collision guard (deletes a leftover `_patch2_backup_*` before `Directory.Move`) — `Patch2GitService.cs:120-124`.
- Restore-on-failure + delete-backup-on-success for the git-exit-code failure path — `Patch2GitService.cs:130-152` / `154-166`.
- Single-flight guard (`_gate`/`_running`, `Patch2GitService.cs:43-44`) lives on the **public** `InitializeOrUpdate` (line 64), acquired/released via `TryEnter`/`finally Exit()` (lines 178, 189) — a concurrent second call correctly returns `AlreadyRunning` (`Patch2GitServiceTests.cs:229-241`).
- `InitializeOrUpdateCore` (internal, line 85) is genuinely pure — no static/guard state, all deps (`isGitRepo`, `cloneOp`, `updateOp`) injected — safe for parallel tests with isolated temp dirs.
- `PatchManagerViewModel.Patch2ButtonText` (`PatchManagerViewModel.cs:93-101`) correctly uses `Patch2GitService.GetPatch2Dir(CoreState.BaseDirectory)` — the repo root — not `ResolvePatchDirectory(version)`, a per-version subfolder that is never itself a git repo.
- UI handlers: synchronous `IsEnabled = false` on click, `Task.Run` off the UI thread, ~150ms-throttled progress via `Dispatcher.UIThread.Post`, and the button is reliably re-enabled in `finally` even when an exception propagates from the service call (`PatchManagerView.axaml.cs:195,211,230-239`; `OptionsView.axaml.cs:183,199,218-226`). No threading hazard.
- Options handler persists only `cfg["submodule_patch2_url"]` + `cfg.Save()` (`OptionsView.axaml.cs:189-195`), never calls `_vm.Save()`. Also checked the empty-textbox edge case: `GitUtil.GetPatch2RemoteUrl()` (`GitUtil.cs:16-24`) falls back to the hardcoded default URL when the config value is blank, so clicking with an emptied textbox degrades gracefully instead of cloning an empty URL.

**Gap** — `FEBuilderGBA.Core/Patch2GitService.cs:129`:
```
int cloneCode = cloneOp(gitExe, url, patchDir, progress, log);
if (cloneCode != 0) { /* restore backup */ }
```
The clone call itself is not wrapped in try/catch. If `cloneOp` (in production `GitUtil.Clone`, which calls `Process.Start`) *throws* rather than returning a non-zero exit code, the previously-moved-aside backup is never restored and the exception just propagates out of `InitializeOrUpdateCore`. This contradicts the class's own doc comment ("restoring the backup on failure", `Patch2GitService.cs:33`) and the "restore-on-failure, no dangling backup" guarantee the rest of the method (and its tests) otherwise deliver.

Impact is bounded, verified by reading the call sites: the single-flight guard is still released (`finally { Exit(); }` in `InitializeOrUpdate`), so no deadlock on retry, and both UI call sites catch `Exception` broadly and re-enable the button in `finally`, so no stuck UI/no crash. Net effect on this narrow path: `config/patch2` is left absent, with the prior tree sitting in an unrestored, timestamped `_patch2_backup_<ticks>` folder until a human notices.

Existing tests only simulate clone failure via `FakeClone.ReturnCode` (`Patch2GitServiceTests.cs:163-181`, `207-226`), never via a thrown `CloneOp`, so this path is untested.

**Requested fix:** wrap the `cloneOp(...)` call at line 129 in try/catch, reuse the existing restore logic (lines 132-138) to roll back the backup, then rethrow (or return `Failed`); add a test with a `CloneOp` fake that throws.

For fairness: the pre-existing WinForms `AutoUpdatePatch2Git` (`FEBuilderGBA/ToolUpdateDialogForm.cs:206-224`), which this PR explicitly and accurately says it "mirrors exactly," has the identical gap (`task.Result` on a faulted clone task throws before the `if (task.Result != 0)` restore block runs) — so this isn't a new pattern invented here. But this Core service is the new tested, canonical implementation shared by two Avalonia call sites (Patch Manager + Options) going forward, so it's worth closing here rather than carrying it forward unfixed.

### 2. Scope
Clean. `git diff origin/master...HEAD --name-only` shows zero files under `FEBuilderGBA/` (WinForms) — genuinely untouched. All 11 changed files are patch2/update-checker related; no FE-Repo/wizard scope creep. The `App.axaml.cs` `--screenshot-window` switch addition for `PatchManagerView` matches an established, repeated pattern (8+ prior PRs), gated behind an explicit CLI arg — not scope creep.

### 3. Tests
Independently re-ran every suite in the fresh, isolated worktree at the exact PR head SHA:
- Core `Patch2GitServiceTests` (10 tests): **10/10 passed**. Deterministic/offline — isolated GUID temp dirs (`Patch2GitServiceTests.cs:16-20`), injected fakes, no network/git dependency. Covers GitNotFound, update success/failure, non-existent-dir clone, pre-existing-empty-dir clone success/fail-restore, pre-existing-non-repo-files clone success/fail-restore, and single-flight `AlreadyRunning`.
- Avalonia `Patch2InitUpdateButtonTests` (4 tests): **4/4 passed** — 2 VM-label tests (repo-root git-status based, using `[Collection("SharedState")]` to safely mutate `CoreState.BaseDirectory`) + 2 headless button-existence tests.
- Regression filter (`PatchManager|Options|ToolUpdateDialog|ControlProperty|AndroidResource`): **90/90 passed**.
- `AvaloniaScrollViewerTests` (structural; lives in `FEBuilderGBA.Core.Tests`, not `.Avalonia.Tests`, despite the name): **3/3 passed**.

All four counts match the PR body's claims exactly. Core + Avalonia Release builds: 0 errors. As noted in §1, the one coverage gap is a "clone throws" scenario, not currently exercised.

### 4. Screenshot
`pr-screenshots/1817/patchmanager-init.png` (master raw URL) resolves successfully as of this review — companion docs PR #1834 has merged. Verified visually: it shows the actual Avalonia Patch Manager empty-state UI with the new "Initialize Patch Database" button — relevant, not a feature-branch URL.

### 5. GUI Test Report
Present — a real 5-row pass/fail table, not boilerplate.

### 6. Test plan
All 7 items checked `[x]`, no placeholders.

---

**Summary:** every plan-board item is correctly implemented except one exception-safety edge case in the new Core service's clone path (§1). It's a small, single-location, mechanical fix with an obvious accompanying test. Once addressed, this is ready to merge.

---
Copilot CLI: 1.0.69-0 / Model: Claude (Anthropic)

File: FEBuilderGBA.Core/Patch2GitService.cs [hunk 1]
```diff
@@ -0,0 +1,194 @@
...
+
+                // Guard against a stale backup directory from a previously aborted run so
+                // Directory.Move never throws on an existing destination.
+                if (Directory.Exists(backupPath))
+                    Directory.Delete(backupPath, true);
+
+                Directory.Move(patchDir, backupPath);
+            }
+
+            int cloneCode = cloneOp(gitExe, url, patchDir, progress, log);
```
[the remark this answers is not in the document]
Comment (laqieer) [reply FEBuilderGBA.Core/Patch2GitService.cs]: Fixed in ba1036a7. `InitializeOrUpdateCore` now wraps **both** `cloneOp` and `updateOp` in try/catch: on a thrown git op it appends the exception to the log and returns a `Failed` result (and, on the clone path, restores the backed-up directory via `RestoreBackup`) — so `config/patch2` is never left stranded under `_patch2_backup_*` and no exception escapes the service. Added `PreExistingDir_CloneThrows_RestoresBackupAndReturnsFailed` and `ExistingRepo_UpdateThrows_ReturnsFailedNoThrow` tests (12 Core tests pass).

Copilot CLI: 1.0.69-0
Model: Claude Opus 4.8 (claude-opus-4.8)

**Comment by laqieer** [comment]
## Cross-Model Review Board — PR #1835

Convened the in-session board (Branch B). All three models plus the `copilot-pull-request-reviewer` bot reviewed the PR; they **converge on a single blocking issue**, now resolved.

| Reviewer | Verdict | Finding |
|---|---|---|
| gemini-3.1-pro-preview | ✅ APPROVE | No bugs; all 8 required fixes + rubric verified |
| gpt-5.5 | 🔴 REQUEST CHANGES | `cloneOp` throw strands `config/patch2` under `_patch2_backup_*` |
| claude-sonnet-5 | 🔴 REQUEST CHANGES | **Same** `cloneOp` throw gap; everything else verified correct (re-ran 10/10, 4/4, 90/90, 3/3 in a clean worktree; screenshot relevant + resolved; GUI Test Report present; test plan fully `[x]`) |
| copilot bot (inline) | 🔴 | **Same** `Patch2GitService.cs:129` throw gap (thread resolved) |

*(gemini/claude noted GitHub blocks `--approve`/`--request-changes` on one's own PR, so verdicts were posted as review comments with the state stated explicitly.)*

### Resolution — commit `ba1036a7`

**1. Exception-safety (the unanimous blocking finding).** `InitializeOrUpdateCore` now wraps **both** `cloneOp` and `updateOp` in try/catch: a thrown git op appends the exception to the log and returns a `Failed` result — and on the clone path restores the backed-up directory via `RestoreBackup` first — so `config/patch2` is never left stranded and no exception escapes the service. New tests: `PreExistingDir_CloneThrows_RestoresBackupAndReturnsFailed`, `ExistingRepo_UpdateThrows_ReturnsFailedNoThrow` → **12/12 Core** pass.

**2. L10n CI gate (separate CI failure).** The full CI suite flagged `L10nCoverageTest.AvaloniaViews_HaveJaAndZhTranslations` — my new AXAML literals lacked ja/zh translations. Added translations for `Initialize Patch Database`, `Initialize / Update Patch Database Now`, and the Update Checker notice (simplified to drop curly-quote/arrow chars for a clean translation key) to `config/translate/{ja,zh}.txt` → **L10n 3/3** pass.

Everything else in the PR was verified correct by all reviewers (single-flight guard placement + core purity, repo-root vs per-version path, UI threading discipline, Options config-key-only persistence, WinForms-untouched scope). The single converging blocking issue is resolved; proceeding to merge once CI is green.

Copilot CLI: 1.0.69-0
Model: Claude Opus 4.8 (claude-opus-4.8)

**Review by laqieer — COMMENTED** [review]
## Updated Verdict: APPROVE

Follow-up to my previous review (comment posted 2026-07-05T01:02:32Z). The sole blocking finding there — `InitializeOrUpdateCore` not restoring the backup when `cloneOp` *throws* (only handled non-zero exit codes) — is now fixed in commit `ba1036a77` (pushed to this PR branch, head is now `ba1036a776dc160a2a6ae6aa580102c44dbd621a`). I verified this independently, not just from the commit message:

**The fix**, `FEBuilderGBA.Core/Patch2GitService.cs`:
- Update path (`code = updateOp(...)`, line 102) is now wrapped in try/catch (line 104); on exception it logs the exception and returns `Failed` (ExitCode -1) — correctly does not attempt a restore, since the update path never moves anything aside.
- Clone path (`cloneCode = cloneOp(...)`, line 150) is now wrapped in try/catch (line 152); on exception it logs the exception, calls the new shared `RestoreBackup` helper (line 158), and returns `Failed` rather than rethrowing. This is actually a better shape than a rethrow — every caller already switches on `result.Kind`, so both the exit-code-failure path and the throw path now go through the identical `Failed` branch, rather than needing two different error-handling shapes.
- The pre-existing non-zero-exit-code restore logic was extracted into the same `RestoreBackup` helper (line 206, called again at line 171) — a pure refactor, no behavior change to the path that was already correct and tested.
- Two new tests directly targeting this: `PreExistingDir_CloneThrows_RestoresBackupAndReturnsFailed` (throwing `CloneOp`, asserts `Failed`, original stray file restored byte-for-byte, zero dangling backups) and `ExistingRepo_UpdateThrows_ReturnsFailedNoThrow` (throwing `UpdateOp`, asserts `Failed`, no throw escapes). Both are deterministic/offline (injected throwing lambdas, isolated GUID temp dirs), exactly like the rest of the suite.
- Independently re-ran the Core suite against this exact commit in a fresh, isolated worktree: **12/12 `Patch2GitServiceTests` passed** (10 original + 2 new).

**Accompanying changes in the same commit** (`ToolUpdateDialogView.axaml`, `config/translate/ja.txt`, `config/translate/zh.txt`): the Update Checker notice text was simplified (dropped curly quotes/#1804 reference) and ja/zh translations were added for the 3 new UI strings this PR introduces ("Initialize Patch Database", "Initialize / Update Patch Database Now", the notice text). This is a necessary, in-scope L10n follow-up, not scope creep — it only touches strings this PR itself introduced.

**Caveat (transparency, not a blocker):** I was not able to complete a final independent re-run of the full Avalonia test suite against this exact commit — the isolated worktree I created for that (`_wt\review1835-fix2`) was removed by another process partway through, evidently a concurrent agent/cleanup routine in this shared environment. I did not recreate it a third time per the instruction to stop build/test loops. Confidence this doesn't matter: the only Avalonia file touched is a static XAML text literal with no control/binding changes, and I already independently verified the full Avalonia regression suite (90/90) plus `Patch2InitUpdateButtonTests` (4/4) against the base commit earlier in this review, none of which assert on this specific notice string.

Screenshot: confirmed independently earlier — `pr-screenshots/1817/patchmanager-init.png` resolves (200) now that #1834 is merged, and shows the actual Patch Manager UI with the new button.

Nothing else outstanding beyond what my previous review already covered (scope clean, single-flight guard correctly placed/pure core, repo-root vs. per-version path usage correct, UI threading/re-enable-on-exception correct, Options config-key-only persistence correct, GUI Test Report present, Test plan fully checked). With the exception-safety fix now landed and verified, this is ready to merge.

---
Copilot CLI: 1.0.69-0 / Model: Claude (Anthropic)
